tools/pythonpkg/duckdb/experimental/spark/sql/session.py

```diff
@@ -168,6 +163,20 @@ def createDataFrame(
             is_empty = True
             data = [tuple(None for _ in names)]
 
+        if not is_empty:
+            if not isinstance(data, list):
+                data = list(data)
+
+            if isinstance(data[0], dict):
+                if not names:
+                    keys = [key for row in data[:10] for key in row.keys()]
```

Comment:
Any reason for this arbitrary limit?
If we have a limit then we should have handling forhitting this limit and corresponding tests for that case

Reply:
The only reason in fact is not iterate over all dataframe, but looking now, this doesn’t make so much effect beacuse the list is already all in memory.